python/mxnet/module/executor_group.py

```diff
@@ -651,6 +652,20 @@ def _bind_ith_exec(self, i, data_shapes, label_shapes, shared_group):
             input_shapes.update(dict(label_shapes))
 
         input_types = {x.name: x.dtype for x in data_shapes}
+        attr_dict = self.symbol.attr_dict()
+
+        for sym_name in self.symbol.list_arguments():
```

Comment:
Could this introduce regressions? 

Can we fold the nested ifs using and or better a function that encapsulates the criteria and makes this easier to read?

Reply:
this is actually improving AMP support for the module api. For example after amp_cast nodes are added dtypes of params won't be able to be inferred easily, so before we provide a default we check the symbol itself if it has a dtype and provide that info to simple_bind. I can refactor to make it easier to read and also think if I can add some test for this.

Reply:
i have added a test for this.